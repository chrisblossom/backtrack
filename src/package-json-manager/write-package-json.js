/* @flow */

import path from 'path';
import { writeFileSync } from 'fs';
import { rootPath } from '../config/paths';
import log from '../utils/log';

import type { PackageJson } from '../types.js';

function writePackageJson(packageJson: PackageJson) {
    const packageJsonFile = path.resolve(rootPath, 'package.json');

    writeFileSync(
        packageJsonFile,
        // eslint-disable-next-line prefer-template
        JSON.stringify(packageJson, null, 2) + '\n',
    );

    log.info('package.json updated');
}

export { writePackageJson };
