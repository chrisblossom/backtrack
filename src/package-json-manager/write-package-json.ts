import path from 'path';
import { writeFileSync } from 'fs';
import { rootPath } from '../config/paths';
import log from '../utils/log';

import { PackageJson } from '../types';

function writePackageJson(packageJson: PackageJson) {
    const packageJsonFile = path.resolve(rootPath, 'package.json');

    writeFileSync(
        packageJsonFile,
        // eslint-disable-next-line prefer-template
        JSON.stringify(packageJson, null, 2) + '\n',
    );

    log.info('package.json updated');
}

export { writePackageJson };
