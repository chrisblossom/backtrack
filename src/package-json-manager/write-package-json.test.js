/* @flow */

import path from 'path';
import fs from 'fs';

const writePackageJson = (packageJson) =>
    require('./write-package-json').writePackageJson(packageJson);

describe('writePackageJson', () => {
    const cwd = process.cwd();
    let writeFileSync;

    beforeEach(() => {
        jest.mock('../utils/log', () => ({
            warn: jest.fn(),
            info: jest.fn(),
            error: jest.fn(),
            success: jest.fn(),
        }));

        /**
         * Before calling '.toMatchSnapshot();' snapshot, call 'writeFileSync.mockRestore();'
         * otherwise snapshot will not write
         */
        writeFileSync = jest
            .spyOn(fs, 'writeFileSync')
            .mockImplementation(() => jest.fn());
    });

    afterEach(() => {
        process.chdir(cwd);
    });

    test('writes package json', () => {
        const dir = path.resolve(__dirname, '__sandbox__/package-json-1/');
        process.chdir(dir);

        const fakePackageJson = {
            scripts: {
                dev: 'run dev',
            },
        };

        const result = writePackageJson(fakePackageJson);

        writeFileSync.mockRestore();

        expect(writeFileSync.mock.calls).toMatchSnapshot();
        expect(result).toEqual(undefined);
    });
});
