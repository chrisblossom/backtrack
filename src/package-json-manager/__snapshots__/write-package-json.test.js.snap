// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writePackageJson writes package json 1`] = `
Array [
  Array [
    "<PROJECT_ROOT>/package.json",
    "{
  \\"scripts\\": {
    \\"dev\\": \\"run dev\\"
  }
}
",
  ],
]
`;
